server/repository/src/migrations/v2_09_00/reintegrate_clinician_gender.rs

```diff
@@ -0,0 +1,96 @@
+use anyhow::Context;
+use diesel::prelude::*;
+use serde::{
+    de::{value::StrDeserializer, IntoDeserializer},
+    Deserialize, Deserializer, Serialize,
+};
+
+use crate::{migrations::*, sync_buffer::sync_buffer, GenderType, SyncAction};
+
+pub fn empty_str_as_option_string<'de, D: Deserializer<'de>>(
+    d: D,
+) -> Result<Option<String>, D::Error> {
+    let s: Option<String> = Option::deserialize(d)?;
+    Ok(s.filter(|s| !s.is_empty()))
+}
+
+pub fn empty_str_as_option<'de, T: Deserialize<'de>, D: Deserializer<'de>>(
+    d: D,
+) -> Result<Option<T>, D::Error> {
+    let s: Option<String> = empty_str_as_option_string(d)?;
+
+    let Some(s) = s else { return Ok(None) };
+
+    let str_d: StrDeserializer<D::Error> = s.as_str().into_deserializer();
+    Ok(Some(T::deserialize(str_d)?))
+}
+
+pub fn ok_or_none<'de, T: Deserialize<'de>, D: Deserializer<'de>>(
+    d: D,
+) -> Result<Option<T>, D::Error> {
+    Ok(empty_str_as_option(d).map_or(None, |v| v))
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct ClinicianOmsFields {
+    #[serde(default)]
+    #[serde(deserialize_with = "ok_or_none")]
+    pub gender: Option<GenderType>,
+}
+
+#[derive(Deserialize, Serialize)]
+struct LegacyClinicianRow {
+    female: bool,
+    oms_fields: Option<ClinicianOmsFields>,
+}
+
+table! {
+  clinician (id) {
+    id -> Text,
+    gender -> Nullable<crate::GenderTypeMapping>,
+  }
+}
+
+#[derive(Clone, Queryable, Insertable, AsChangeset, Debug, PartialEq, Default)]
+#[diesel(table_name = clinician)]
+pub struct ClinicianRow {
+    pub id: String,
+    pub gender: Option<GenderType>,
+}
```

Comment:
Similarly these are all duplicated/subsets of other types that exist and is quite nasty IMO. It makes me want to review our project structure entirely 👀. One might be able to argue that the problem is that I should be using the repository layer rather than diesel directly, however that is not the established pattern in migration `templates` directory. That also wouldn't fix the lack of access to `LegacyClinicianRow` and `ClinicianOmsFields`. Maybe there's a case that it's better this way?